python/ray/serve/tests/test_handle.py

```diff
@@ -353,6 +353,19 @@ async def __call__(self, name: str):
     assert ray.get(h.remote("sned")) == "Hi sned"
 
 
+def test_handle_options_with_same_router(serve_instance):
+    """Make sure that multiple handles share same router object."""
+
+    @serve.deployment
+    def echo(name: str):
+        return f"Hi {name}"
+
+    handle = serve.run(echo.bind())
+    handle2 = handle.options(multiplexed_model_id="model2")
+    assert handle._router
+    assert id(handle2._router) == id(handle._router)
```

Comment:
I don't think this test fails without your change -- both would be `None` and they'd have the same id

Reply:
it is failing because i have `assert handle._router` in the previous line.

Reply:
Hmm I don't like that this test is basically just verifying implementation details

Reply:
if i understand you correctly, i can use existing api `_get_or_create_router` from handle to verify the same sharing router object.

Reply:
well, `_get_or_create_router` is also an implementation detail...

Reply:
In general it's best if tests only use public interfaces. It ensures that you are testing the end-user behavior and avoids pain when you refactor implementation details.

In this case I can't really think of a better alternative unfortunately.